### PR TITLE
Add optional autoincrement function expression to SQLDialect

### PR DESCRIPTION
**Pull request by johan-carlberg**
Some databases doesn’t support the `AUTOINCREMENT` clause, but have the ability to implement autoincrementing primary keys using a function as a default instead.

### CODE REVIEW

File: Sources/SQLKit/SQLDialect.swift [hunk 1]
```diff
@@ -8,6 +8,8 @@ public protocol SQLDialect {
     var literalDefault: SQLExpression { get }
     var supportsIfExists: Bool { get }
     var supportsAutoIncrement: Bool { get }
+    var supportsAutoIncrementUsingDefaultFunction: Bool { get }
+    var autoIncrementFunction: SQLExpression { get }
```
Comment (tanner0101) [inline Sources/SQLKit/SQLDialect.swift]: These two properties could be combined by having `autoIncrementFunction` be optional (`SQLExpression?`).

File: Sources/SQLKit/Query/SQLColumnConstraintAlgorithm.swift [hunk 2]
```diff
@@ -92,12 +92,20 @@ public enum SQLColumnConstraintAlgorithm: SQLExpression {
     public func serialize(to serializer: inout SQLSerializer) {
         switch self {
         case .primaryKey(let autoIncrement):
+            if autoIncrement {
+                if serializer.database.dialect.supportsAutoIncrementUsingDefaultFunction{
```
Comment (tanner0101) [inline Sources/SQLKit/Query/SQLColumnConstraintAlgorithm.swift]: What do you think about this structure:
```swift
if serializer.database.dialect.supportsAutoIncrement {
    if let function = serializer.database.dialect.autoIncrementFunction {
        // serialize DEFAULT + function
    } else {
        // serialize autoIncrementClause
    }
} else {
    // warn not supported 
}
```

File: Sources/SQLKit/SQLDialect.swift (hunk 1 above)
Comment (johan-carlberg) [reply Sources/SQLKit/SQLDialect.swift]: Ah, yes, that’s kind of obvious. Will fix.

File: Sources/SQLKit/Query/SQLColumnConstraintAlgorithm.swift (hunk 2 above)
Comment (johan-carlberg) [reply Sources/SQLKit/Query/SQLColumnConstraintAlgorithm.swift]: Will probably make the code cleaner, so I’ll try to make it so.

**Comment by tanner0101** [comment]
@johan-carlberg looks like this one also has some merge conflicts I'm not able to resolve on my end. Are you able to resolve those?

**Comment by johan-carlberg** [comment]
> @johan-carlberg looks like this one also has some merge conflicts I'm not able to resolve on my end. Are you able to resolve those?

Done!

**Comment by tanner0101** [comment]
@johan-carlberg thank you for fixing, merging your other PR created another conflict though 😓. I've invited you to the vapor/contributors team so future PRs can be on a local branch and I can help fix these.

**Comment by johan-carlberg** [comment]
Ok, I’ll have a look.

**Comment by tanner0101** [comment]
These changes are now available in [3.0.0-beta.6.5](https://github.com/vapor/sql-kit/releases/tag/3.0.0-beta.6.5)

**Comment by tanner0101** [comment]
Thanks for the quick fix @johan-carlberg

**Pull request merged**